src/KGzocha/Searcher/Searcher/Searcher.php

```diff
@@ -41,7 +42,7 @@ public function search(
             $this->searchForModel($filterModel, $this->searchingContext);
         }
 
-        return $this->searchingContext->getResults();
+        return new ResultCollection($this->searchingContext->getResults());
```

Comment:
Worth consideration to extract this to protected method, so it would look like:
```php
return $this->wrapResuts($this->searchingContext->getResults());
```
and then we add:
```php
protected function wrapResults($results)
{
    return new ResultCollection($results);
}
```
In this way it will be a lot easier to extend Searcher itself and change `ResultCollection`